routes/errortracker.js

```diff
@@ -0,0 +1,257 @@
+/**
+ * Created by rigwanzojunior on 6/5/17.
+ *objects used
+ *Objects are identical to the objects used in previous errortracker.go
+ * type errorRequestMeta {
+	HTTPReferrer  string `json:"http_referrer,omitempty"`
+	HTTPUserAgent string `json:"http_user_agent,omitempty"`
+    }
+ type errorRequest {
+	URL    string            `json:"url,omitempty"`
+	Method string            `json:"method,omitempty"`
+	Meta   *errorRequestMeta `json:"meta,omitempty"`
+ }
+ type errorEvent {
+	Application string `json:"application,omitempty"`
+	AppID       string `json:"app_id,omitempty"`
+	Environment string `json:"environment,omitempty"`
+	Version     string `json:"version,omitempty"`
+
+	Message   string `json:"message,omitempty"`
+	Exception string `json:"exception,omitempty"`
+
+	Request *errorRequest `json:"request,omitempty"`
+
+	Filename  string `json:"filename,omitempty"`
+	Line      int32  `json:"line,omitempty"`
+	Classname string `json:"classname,omitempty"`
+	Function  string `json:"function,omitempty"`
+	Severity  string `json:"severity,omitempty"`
+ }
+ * Handle error requests from clients and log them.
+ */
+//AppEngine project ID
+const projectId ='';
+const logName = 'javascript.errors';
+const express = require('express');
+const router = express.Router();
+const logging = require('@google-cloud/logging');
+const winston = require('winston');
+const statusCodes = require('http-status-codes');
+const url = require('url');
+//String constants for identifying error level.
+const errorLevels = {
+    Default:   "Default",
+    Debug:     "Debug",
+    Info:      "Info",
+    Notice:    "Notice",
+    Warning:   "Warning",
+    Error:     "Error",
+    Critical:  "Critical",
+    Alert:     "Alert",
+    Emergency: "Emergency"
+};
+//params in http request
+var params = {};
+
+/**
+ * Filter exceptions in an array to prevent them from being logged
+ * @param errorEvent
+ * @returns {boolean}
+ */
+function isFilteredMessageOrException(errorEvent) {
+    var filteredMessageOrException = ["stop_youtube",
+        "null%20is%20not%20an%20object%20(evaluating%20%27elt.parentNode%27)"];
+    filteredMessageOrException.forEach(function filter(msg){
+        if(errorEvent.Message.toString().includes(msg) || errorEvent.Exception.toString().includes(msg)){
+            return true;
```

Comment:
This return does not return from `isFilteredMessageOrException`, but rather this iterating `#forEach` function. Instead, we can use a `#some`:

```js
return filteredErrors.some((msg) => {
  return errorEvent.Message.toString().includes(msg) || ...
});